### PR TITLE
[fix](function) make TIMESTAMP function DEPEND_ON_ARGUMENT

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes

before
```
select millisecond_timestamp(cast('2023-11-18 00:09:32' as datetime));
function return type check failed, function_name=millisecond_timestamp, expect_return_type=Int64, 

real_return_type=Nullable(Int64), input_arguments=DateTimeV2
```

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

**Review by zclllyybb — CHANGES_REQUESTED** [review]
Please change this function to `DEPEND_ON_ARGUMENT` when it has no stringlike args. no need to use `always_nullable`.

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

File: be/src/vec/functions/function_timestamp.cpp [hunk 1]
```diff
@@ -804,39 +804,63 @@ class DateTimeToTimestamp : public IFunction {
...
-        res_col->get_data().resize_fill(input_rows_count, 0);
-        null_vector->get_data().resize_fill(input_rows_count, false);
-        NullMap& null_map = null_vector->get_data();
-        auto& res_data = res_col->get_data();
-        const cctz::time_zone& time_zone = context->state()->timezone_obj();
-        for (int i = 0; i < input_rows_count; i++) {
-            if (arg_col->is_null_at(i)) {
-                null_map[i] = true;
-                continue;
+        if (arg_col->is_nullable()) {
```
Comment (zclllyybb) [inline be/src/vec/functions/function_timestamp.cpp:816]: we have a framework to deal when all arguments are nullable. Is this necessary?

**Comment by Mryange** [comment]
run buildall

**Review by zclllyybb — APPROVED** [review]
LGTM

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run pipelinex_p0

**Comment by Mryange** [comment]
run buildall

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**